pkg/controller/nginxingress/nginx_ingress_controller.go

```diff
@@ -581,6 +583,14 @@ func ToNginxIngressConfig(nic *approutingv1alpha1.NginxIngressController, defaul
 		nginxIng.DefaultBackendService = nic.Spec.DefaultBackendService.Namespace + "/" + nic.Spec.DefaultBackendService.Name
 	}
 
+	if len(nic.Spec.CustomHTTPErrors) > 0 {
+		errAsStr := make([]string, len(nic.Spec.CustomHTTPErrors))
+		for _, errCode := range nic.Spec.CustomHTTPErrors {
+			errAsStr = append(errAsStr, strconv.Itoa(errCode))
```

Comment:
this block is incorrect. when you use make you are initializing the slice you are setting the length to the length of customHttpErrors then adding additional fields.

https://go.dev/play/p/1GFim9dRtyM

You can instead set the length of the slice with a length of 0 and capacity of the expected length. or you could index into it

We need to have a unit test for this block that can prove this block works

Reply:
would recommend indexing for ideal performance so you don't have to continually reallocate memory with an append like you would if you started with a size/capacity of 0

Reply:
Ha that was my bad in a previous comment, should index

Reply:
Reverted it and the test already exists in nginx_ingress_controller_test.go @ line 1338 if I'm not mistaken